### PR TITLE
Add detailed debug logging in executeBlock

### PR DESCRIPTION
**Pull request by Kay-Zee**

### CODE REVIEW

**Review by SaveTheRbtz — APPROVED** [review]

File: engine/execution/computation/computer/computer.go [hunk 1]
```diff
@@ -196,12 +204,25 @@ func (e *blockComputer) executeBlock(
 		if err != nil {
 			return nil, fmt.Errorf("failed to execute collection at txIndex %v: %w", txIndex, err)
 		}
+		collectionID := collection.Guarantee.CollectionID[:]
```
Comment (SaveTheRbtz) [inline engine/execution/computation/computer/computer.go:207]: you may consider creating collectionLogger here.

File: engine/execution/computation/computer/computer.go [hunk 2]
```diff
@@ -196,12 +204,25 @@ func (e *blockComputer) executeBlock(
...
+			Msg("collection events hashed")
 		err = stateView.MergeView(colView)
 		if err != nil {
 			return nil, fmt.Errorf("cannot merge view: %w", err)
 		}
+		blockLogger.Debug().
+			Hex("collection_id", collectionID).
+			Msg("collection view merged")
 		collectionIndex++
 	}
```
Comment (SaveTheRbtz) [inline engine/execution/computation/computer/computer.go:228]: maybe also something after the execution of the system collection

**Review by SaveTheRbtz — APPROVED** [review]

File: engine/execution/computation/computer/computer.go [hunk 3]
```diff
@@ -196,12 +204,25 @@ func (e *blockComputer) executeBlock(
...
+			Hex("collection_id", collectionID).
+			Msg("collection executed")
 		bc.Commit(colView)
+		blockLogger.Debug().
+			Hex("collection_id", collectionID).
+			Msg("collection committed")
 		eh.Hash(res.Events[i])
+		blockLogger.Debug().
+			Hex("collection_id", collectionID).
+			Msg("collection events hashed")
```
Comment (SaveTheRbtz) [inline engine/execution/computation/computer/computer.go:218]: these maybe a bit too much, given that both Commit and Hash are async.

Comment (smnzhu) [reply engine/execution/computation/computer/computer.go:218]: I think the more accurate log line would be something like "collection events hashing initiated". In the callback, we can say "collection events hashed"

File: engine/execution/computation/computer/computer.go [hunk 4]
```diff
@@ -175,17 +179,21 @@ func (e *blockComputer) executeBlock(
 				panic(err)
 			}
 			res.EventsHashes = append(res.EventsHashes, hash)
+			blockLogger.Debug().Msg("event hashed (callback)")
```
Comment (smnzhu) [inline engine/execution/computation/computer/computer.go:182]: Would itbbe worth adding the collection ID to the callback and logging it so it's easier to correlate the two debug lines before and after hashing? Either that or the index

**Review by smnzhu — APPROVED** [review]

**Comment by Kay-Zee** [comment]
going to close this, but feel free to still use it as an example if needed @simonhf

**Pull request closed without merging**